### DISCUSSION TITLE
How do I generate links into Sentry settings without knowing the organisation slug?

### DISCUSSION THREAD

**Discussion by lawrencejones** [getsentry/support#6] [Support] [answered]
Hi all!

We're building a Sentry integration, and want to provide a link to "Connect to Sentry" from within our app.

For all our other integrations, we can generate a link that takes the user to the "Authorise/OAuth flow/Connect" page in the relevant integration, and they can pick the account they want to authorise.

For Sentry, I'm having an issue figuring out how to generate that link.

I know I can link users to:

```
https://sentry.io/settings/<org-slug>/sentry-apps/<app-name>/
```

While I know the app name, I need to ask users for their organisation slug/ID in a form before I can generate them a link.

Is that a requirement? I worry people won't know what an organisation slug is, and may mistype things and get sent to a 404 page.

In an ideal world, there's an organisation agnostic endpoint that captures the request, allows someone to sign into their relevant Sentry account, and forwards them onto the original path once they sign in.

Does this exist?

Thanks!

**Comment by scefali** [comment, marked as answer]
@lawrencejones Yes, this exists but is not documented. You can use the URL `https://sentry.io/sentry-apps/:sentryAppSlug/external-install/` where `:sentryAppSlug` is the slug of your integration. Example for Shortcut: https://sentry.io/sentry-apps/shortcut/external-install/. If the user has multiple organizations they will be presented with a selector to choose it, otherwise it will default to the one they are a member of.

**Comment by lawrencejones** [comment]
Thanks for this, was really useful!
